src/mem/ruby/system/GPUCoalescer.cc

```diff
@@ -599,6 +599,10 @@ GPUCoalescer::hitCallback(CoalescedRequest* crequest,
                 // data response is not needed.
                 case RubyRequestType_ATOMIC_NO_RETURN:
                     assert(pkt->isAtomicOp());
+                    log = data.popAtomicLogEntryFront();
+                    delete [] log;
```

Comment:
Sorry to suggest yet another change, especially since I suggest this.  However, I noticed that clearAtomicLogEntries() will delete *all* of the entries and clear the log.  The code I suggested assumes that the log only has one entry which I think is not necessarily true:

`data.clearAtomicLogEntries()`

Reply:
This is already within a for-loop -- wouldn't that take care of all the log entries?

Reply:
Oh I see.  The relation between DataBlock and packet was a bit unclear until I read the comment for atomic return.  This look fine to me then